RA-10

KinCadeFoster opened the pull request:
- Во фрагменте с категориями в метод openRecipesByCategoryId() добавил принимающий целочисленный параметр categoryId
- Передавал этот параметр из CategoriesListAdapter через коллбэк
- В методе openRecipesByCategoryId() создал две переменные categoryName и categoryImageUrl
- В методе openRecipesByCategoryId() создал переменную bundle
- Передавал созданный объект Bundle в метод замены фрагмента в качестве параметра args
- Создал 3 переменные для принимаемых данных и проинициализировал их null
- В методе onViewCreated() получаю передаваемые данные из аргументов

KinCadeFoster committed:
RA-10
- Во фрагменте с категориями в метод openRecipesByCategoryId() добавил принимающий целочисленный параметр categoryId
- Передавал этот параметр из CategoriesListAdapter через коллбэк
- В методе openRecipesByCategoryId() создал две переменные categoryName и categoryImageUrl
- В методе openRecipesByCategoryId() создал переменную bundle
- Передавал созданный объект Bundle в метод замены фрагмента в качестве параметра args
- Создал 3 переменные для принимаемых данных и проинициализировал их null
- В методе onViewCreated() получаю передаваемые данные из аргументов

ievetrovdev commented on app/src/main/java/ru/example/recipeapp/CategoriesListFragment.kt:
```diff
@@ -36,17 +36,29 @@ class CategoriesListFragment : Fragment() {
-            addToBackStack(null)
+        val category = STUB.getCategories().find { it.id == categoryId }
+
+        if (category != null) {
```
Молодец, что перехватываешь ситуацию с null.  Можно сделать больше в стиле Котлин через:
category?.let{ 
}

ievetrovdev commented on app/src/main/res/layout/fragment_list_recipes.xml:
```diff
@@ -9,6 +9,7 @@
     android:background="@color/main_background_color">
 
     <TextView
+        android:id="@+id/textView4"
```
Понятно, что тестовая вьюха. Но лучше привыкать давать осмысленные названия в едином стиле

ievetrovdev reviewed (approved):
Всё корректно, молодец! Пару общих комментов оставил

ievetrovdev commented on app/src/main/java/ru/example/recipeapp/CategoriesListFragment.kt:
```diff
@@ -36,17 +36,29 @@ class CategoriesListFragment : Fragment() {
+            val categoryImageUrl = category.imageUrl
+
+            val bundle = bundleOf(
+                "ARG_CATEGORY_ID" to category.id,
```
Кючи для аргументов принято выносить общие константы проекта в отдельный файл. Его можно назвать, например, Constants.kt и переиспользовать оттуда в разных местах, где это требуется.

KinCadeFoster committed:
RA-10
- переработал под category?.let
- вынес константы в Constants.kt
- переименовал id textView

ievetrovdev reviewed (approved):
Молодец! Всё корректно сделал
